Source/SwiftLintFramework/Models/Configuration.swift

```diff
@@ -78,17 +79,33 @@ public struct Configuration: Equatable {
         }
         self.disabledRules = validDisabledRules
 
-        self.rules = rules.filter { rule in
-            let id = rule.dynamicType.description.identifier
-            if validDisabledRules.contains(id) { return false }
-            return enabledRules.contains(id) || !(rule is OptInRule)
+        // Whitelist rules take precendence over all else.
+        if !whitelistRules.isEmpty {
+
+            // Error if enabled_rules or disabled_rules is set while whitelist_rules are.
+            if !disabledRules.isEmpty || !optInRules.isEmpty {
+                queuedPrintError("Other rulesets can not exist alongside whitelist_rules")
```

Comment:
I think explicitly referencing `disabled_rules` and `opt_in_rules` would be clearer:

> 'disabled_rules' or 'opt_in_rules' cannot be used in combination with 'whitelist_rules'